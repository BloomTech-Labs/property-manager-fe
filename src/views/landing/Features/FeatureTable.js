import React from 'react';
import features from './FeatureList';
import FeatureButtons from './FeatureButtons';
import './features.scss';

const FeatureTable = () => {
  return (
    <div className="featureTableWrapper">
      <table className="featureTable">
        <thead className="featureTiersWrapper">
          <tr className="featureTiers">
            <th />
            <th>Free</th>
            <th>$49/mo</th>
            <th>$99/mo</th>
          </tr>
        </thead>
        <tbody className="featureTableBodyWrapper">
          {features.map(({ feature, free, middle, premium }) => (
            <tr key={feature} className="tableRow">
              <td className="featureName">{feature}</td>
              <td>{free === true ? '✔' : ''}</td>
              <td>{middle === true ? '✔' : ''}</td>
              <td>{premium === true ? '✔' : ''}</td>
            </tr>
          ))}
        </tbody>
      </table>
      <FeatureButtons />
    </div>
  );
};

export default FeatureTable;
