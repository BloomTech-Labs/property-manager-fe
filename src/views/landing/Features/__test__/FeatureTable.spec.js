import React from 'react';
import * as rtl from '@testing-library/react';
import '@testing-library/jest-dom';
import FeatureTable from '../FeatureTable';

afterEach(rtl.cleanup);

test('Makes sure the word free exists on LP', async () => {
  const wrapper = rtl.render(<FeatureTable />);
  expect(await wrapper.queryAllByText(/free/i));
});

test('snapshot', () => {
  const { container } = rtl.render(<FeatureTable />);
  expect(container.firstChild).toMatchSnapshot();
});
