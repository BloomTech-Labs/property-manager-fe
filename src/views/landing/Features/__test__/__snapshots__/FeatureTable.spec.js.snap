// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<div
  class="featureTableWrapper"
>
  <table
    class="featureTable"
  >
    <thead
      class="featureTiersWrapper"
    >
      <tr
        class="featureTiers"
      >
        <th />
        <th>
          Free
        </th>
        <th>
          $49/mo
        </th>
        <th>
          $99/mo
        </th>
      </tr>
    </thead>
    <tbody
      class="featureTableBodyWrapper"
    >
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Professional dashboard
        </td>
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Emailing service
        </td>
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Workorder management
        </td>
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Accept payments
        </td>
        <td />
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Store documents
        </td>
        <td />
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Downloadable applications
        </td>
        <td />
        <td>
          ✔
        </td>
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Premium messaging service
        </td>
        <td />
        <td />
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          24/7 Premium support
        </td>
        <td />
        <td />
        <td>
          ✔
        </td>
      </tr>
      <tr
        class="tableRow"
      >
        <td
          class="featureName"
        >
          Itemized expense tracking
        </td>
        <td />
        <td />
        <td>
          ✔
        </td>
      </tr>
    </tbody>
  </table>
  <div
    class="buy-btn-wrapper"
  >
    <div
      class="empty"
    />
    <button
      class="buy-btn"
      type="submit"
    >
      Buy
    </button>
    <button
      class="buy-btn"
      type="submit"
    >
      Buy
    </button>
    <button
      class="buy-btn"
      type="submit"
    >
      Buy
    </button>
  </div>
</div>
`;
